mne/epochs.py

```diff
@@ -1810,6 +1810,56 @@ def save(self, fname, split_size='2GB', fmt='single', overwrite=False,
             this_epochs.event_id = self.event_id
             _save_split(this_epochs, fname, part_idx, n_parts, fmt)
 
+    def export(self, fname, fmt="auto"):
+        """Export Epochs to EEGLAB's .set format.
+
+        Parameters
+        ----------
+        fname : str
+            Name of the export file.
+        fmt : str
+            Format of the export. Defaults to auto, which will infer
+            format from fname's extension
+
+        Notes
+        -----
+        Channel locations are expanded to the full EEGLAB format
+        For more details see eeglabio.utils.cart_to_eeglab
+        """
+        if fmt == "auto":
+            fmt = op.splitext(fname)[1]
+            if fmt:
+                fmt = fmt[1:]
+            else:
+                raise ValueError("Couldn't infer format from filename "
+                                 "(no extension found)")
+
+        if fmt == "set":
+            from eeglabio.epochs import export_set
+            # load data first
+            self.load_data()
+
+            # remove extra epoc and STI channels
+            chs_drop = \
+                [ch for ch in ['epoc', 'STI 014'] if ch in self.ch_names]
+            self.drop_channels(chs_drop)
```

Comment:
Please don't change the object in-place. You could either make a copy, or even better pick the channels you want in `self.get_data()` a few lines down.

Reply:
Fixed.